File: app/src/main/java/com/droidacid/apticalc/common/AptiApp.java
package com.droidacid.apticalc.common;

import android.app.Application;

/**
 * Created by shivam on 3/9/17.
 */

public class AptiApp extends Application {


    private static AptiApp instance;

    public static AptiApp getInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
    }
}

File: app/src/main/java/com/droidacid/apticalc/common/AptiApp.kt
package com.droidacid.apticalc.common

import android.app.Application
import com.droidacid.apticalc.di.Injection

/**
 * Created by shivam on 3/9/17.
 */

class AptiApp : Application() {

    companion object {
        var instance: AptiApp? = null
            private set
    }

    override fun onCreate() {
        super.onCreate()
        instance = this
    }

    fun getInjection() = Injection(this)
}